### PR TITLE
fix(plan-orchestrate): detect renamed ecc@ecc marketplace install

### LINKED ISSUE #2316: plan-orchestrate: stale ECC install detection after marketplace rename to `ecc@ecc`
`plan-orchestrate` Phase 0 detection still looks for:
- `~/.claude/plugins/marketplaces/everything-claude-code/` (plugin mode)
- `~/.claude/agents/` with ECC agent files (legacy mode)

With the new `ecc@ecc` marketplace identifier (2.0.0+), the install path is
`~/.claude/plugins/marketplaces/ecc/` — neither condition matches, so the skill
always falls back to legacy mode with a warning.

Fix needed: add a third detection case for `<claude-home>/plugins/marketplaces/ecc/`
→ `ECC_MODE=plugin` with `{ORCH_CMD}=/ecc:orchestrate` and `{AGENT(name)}=ecc:<name>`.

Repro: install ECC from marketplace (ecc@ecc), run `/ecc:plan-orchestrate <any-plan>.md`.
Warning: "could not detect ECC install; defaulting to legacy form."

ECC version: 2.0.0 | OS: Linux

### PR DESCRIPTION
**Pull request by gaurav0107**
## What Changed

`plan-orchestrate` Phase 0 install detection (`skills/plan-orchestrate/SKILL.md`) now recognizes the renamed `ecc@ecc` marketplace install and resolves the command/agent prefix from the marketplace namespace:

- Added a detection case for `<claude-home>/plugins/marketplaces/ecc/` → `ECC_MODE=plugin`, `{NS}=ecc` (canonical, 2.0.0+).
- Kept `<claude-home>/plugins/marketplaces/everything-claude-code/` → `{NS}=everything-claude-code` for backward compatibility (pre-2.0.0 installs).
- Parameterized the prefix as a Phase 0 token `{NS}`: `{ORCH_CMD}` = `/{NS}:orchestrate`, `{AGENT(name)}` = `{NS}:<name>` under a plugin install; bare under legacy.
- Mixed-install tie-break prefers `ecc` (the current identifier).
- Updated the detection table, rendering rules, Phase 0 step 5 / Phase 5 self-check / "Plan declares agents" edge case to strip either plugin prefix, and switched the canonical example to `/ecc:orchestrate`.
- Added `tests/docs/plan-orchestrate-install-detection.test.js` to pin the behavior.

## Why This Change

After the 2.0.0 marketplace rename to `ecc@ecc`, the install path is `<claude-home>/plugins/marketplaces/ecc/`. The skill only matched the old `everything-claude-code` marketplace path and the legacy bare install, so it always fell through to `ECC_MODE=legacy` — emitting the spurious warning `could not detect ECC install; defaulting to legacy form` and bare, unprefixed `/orchestrate` invocations that do not resolve under the plugin install. This aligns `plan-orchestrate` with the canonical `/ecc:` namespace the repo already enforces for public docs (`tests/docs/install-identifiers.test.js`).

## Testing Done

- `node tests/docs/plan-orchestrate-install-detection.test.js` — 6/6 pass (new).
- `node tests/run-all.js` — 2943/2943 pass, 0 failures.
- `node scripts/ci/validate-skills.js` — 277 skill directories validated.
- `npx eslint scripts/**/*.js tests/**/*.js` — clean.
- `npx markdownlint skills/plan-orchestrate/SKILL.md` — clean.
- `node scripts/ci/validate-no-personal-paths.js` — clean.
- `node scripts/ci/check-unicode-safety.js` — clean.
- `node scripts/ci/catalog.js --text` — counts match.
- `npm run command-registry:check` — registry up to date.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation update

## Security & Quality Checklist

- [x] Surgical scope — one skill file plus its regression test.
- [x] No secrets, credentials, or personal absolute paths added.
- [x] No generated files hand-edited.
- [x] Backward compatible — pre-2.0.0 plugin and legacy bare installs still detected.
- [x] Conventional Commit message; PR follows the repo template.

## Documentation

The change is to the `plan-orchestrate` skill instructions themselves; the detection table, Phase 0 algorithm, rendering rules, and Example 1 are all updated in-place to document the `ecc@ecc` form.

Fixes #2316

## Summary by CodeRabbit

* **Documentation**
  * Updated the orchestration guide to recognize the canonical ECC plugin namespace and show the correct command and agent-name format.
  * Clarified how names are normalized to avoid duplicate prefixes in plugin mode.
  * Refreshed examples to match the new ECC-based output.

* **Tests**
  * Added regression coverage for ECC install detection and namespace rendering, including legacy compatibility checks.

### CODE REVIEW

**Comment by gaurav0107** [comment]
@coderabbitai review

File: skills/plan-orchestrate/SKILL.md [hunk 1]
```diff
@@ -99,7 +103,7 @@ A misspelled agent name fails `/orchestrate`. Cross-check against this list befo
...
 4. Detect a **PyTorch sub-profile**: when `lang=python` and any of `pyproject.toml` / `requirements.txt` / `uv.lock` declares a dependency on `torch`, set `pytorch=true`. This only affects `build` chain selection (Phase 2 rule below); the reviewer remains `python-reviewer`.
-5. **Normalize any agent names declared in the plan**: if the plan text references agents by their plugin-prefixed form (e.g. `everything-claude-code:tdd-guide`), strip the prefix to get the bare catalogue name before validating or composing chains. Re-prefixing happens only at output time per `ECC_MODE` (Phase 4). Never let a pre-prefixed name flow into chain composition — it would double-prefix in plugin mode.
+5. **Normalize any agent names declared in the plan**: if the plan text references agents by a plugin-prefixed form (e.g. `ecc:tdd-guide` or `everything-claude-code:tdd-guide`), strip the `{NS}:` prefix to get the bare catalogue name before validating or composing chains. Re-prefixing happens only at output time per `ECC_MODE` (Phase 4). Never let a pre-prefixed name flow into chain composition — it would double-prefix in plugin mode.
```
[the remark this answers is not in the document]
Comment (gaurav0107) [reply skills/plan-orchestrate/SKILL.md]: Addressed in 03d1bf9. Phase 0 step 5 now strips any known plugin prefix (`ecc:` or `everything-claude-code:`), not just the detected `{NS}:`, matching the Phase 5 self-check and the "Plan declares agents" edge case. Thanks for the catch.

[the remark this answers is not in the document]
Comment (gaurav0107) [reply skills/plan-orchestrate/SKILL.md]: Addressed in 03d1bf9 — step 5 now strips any known plugin prefix to match lines 210/228. Thanks for the review.

File: tests/docs/plan-orchestrate-install-detection.test.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+test('prefers the ecc marketplace over the legacy id (ordering)', () => {
+  const eccIdx = skill.indexOf('marketplaces/ecc/');
+  const legacyIdx = skill.indexOf('marketplaces/everything-claude-code/');
+  assert.ok(eccIdx !== -1 && legacyIdx !== -1, 'Expected both marketplace paths to be documented');
+  assert.ok(
+    eccIdx < legacyIdx,
+    'Expected the canonical ecc marketplace to be matched before the legacy everything-claude-code marketplace'
+  );
+});
```
[the remark this answers is not in the document]
Comment (gaurav0107) [reply tests/docs/plan-orchestrate-install-detection.test.js:70]: Addressed in 03d1bf9. Added a test that scopes the ordering assertion to the Phase 0 detection-algorithm block (between "Detect ECC install form once and freeze" and "From this point on"), so the algorithm order can no longer regress while the table/examples keep the document-wide check green.

File: tests/docs/plan-orchestrate-install-detection.test.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+test('still documents the legacy bare install and its fallback warning', () => {
+  assert.ok(
+    skill.includes('Legacy bare install'),
+    'Expected the legacy bare install form to remain documented'
+  );
+  assert.ok(
+    skill.includes('could not detect ECC install'),
+    'Expected the legacy-fallback warning to remain documented'
+  );
+});
```
[the remark this answers is not in the document]
Comment (gaurav0107) [reply tests/docs/plan-orchestrate-install-detection.test.js:110]: Addressed in 03d1bf9 — added a regression asserting step 5 normalizes both `ecc:` and `everything-claude-code:` before catalogue validation (checks for the "strip any known plugin prefix" rule plus both prefixed forms).

File: skills/plan-orchestrate/SKILL.md [hunk 4]
```diff
@@ -86,11 +89,12 @@ A misspelled agent name fails `/orchestrate`. Cross-check against this list befo
...
-   2. Else if `<claude-home>/agents/` exists and contains at least one ECC agent file (e.g. `tdd-guide.md`, `code-reviewer.md`) → `ECC_MODE=legacy`.
-   3. Else → default to `ECC_MODE=legacy` and emit a one-line warning at the top of the output: `> Warning: could not detect ECC install; defaulting to legacy form. If you use the plugin install, edit the prefixes manually.`
-   4. If both markers exist (mixed install), `plugin` wins — the plugin namespace is the only one that resolves agent names without fuzzy matching.
+2. Detect ECC install form once and freeze it into `ECC_MODE` and (for plugin installs) the marketplace namespace `{NS}`. Algorithm (run in order, stop at the first match):
+   1. If `<claude-home>/plugins/marketplaces/ecc/` exists → `ECC_MODE=plugin`, `{NS}=ecc` (the canonical `ecc@ecc` marketplace, 2.0.0+).
+   2. Else if `<claude-home>/plugins/marketplaces/everything-claude-code/` exists → `ECC_MODE=plugin`, `{NS}=everything-claude-code` (pre-2.0.0 marketplace id).
+   3. Else if `<claude-home>/agents/` exists and contains at least one ECC agent file (e.g. `tdd-guide.md`, `code-reviewer.md`) → `ECC_MODE=legacy`.
```
[the remark this answers is not in the document]
Comment (gaurav0107) [reply skills/plan-orchestrate/SKILL.md:95]: Thanks — I looked into this but left it out of scope for #2316. Claude Code marketplace plugins install under `<claude-home>/plugins/marketplaces/<marketplace>/`; the issue and the repo's own install-identifiers test pin the path to `plugins/marketplaces/ecc/`, which this change detects. A flat `plugins/ecc/` or `plugins/ecc@ecc/` layout isn't a path Claude Code produces, and the pre-existing detection already only matched `marketplaces/...` forms, so this isn't a regression introduced here. Happy to open a follow-up if a real flat-install layout is confirmed.

File: skills/plan-orchestrate/SKILL.md [hunk 5]
```diff
@@ -86,11 +89,12 @@ A misspelled agent name fails `/orchestrate`. Cross-check against this list befo
...
-   1. If `<claude-home>/plugins/marketplaces/everything-claude-code/` exists → `ECC_MODE=plugin`.
-   2. Else if `<claude-home>/agents/` exists and contains at least one ECC agent file (e.g. `tdd-guide.md`, `code-reviewer.md`) → `ECC_MODE=legacy`.
-   3. Else → default to `ECC_MODE=legacy` and emit a one-line warning at the top of the output: `> Warning: could not detect ECC install; defaulting to legacy form. If you use the plugin install, edit the prefixes manually.`
-   4. If both markers exist (mixed install), `plugin` wins — the plugin namespace is the only one that resolves agent names without fuzzy matching.
+2. Detect ECC install form once and freeze it into `ECC_MODE` and (for plugin installs) the marketplace namespace `{NS}`. Algorithm (run in order, stop at the first match):
+   1. If `<claude-home>/plugins/marketplaces/ecc/` exists → `ECC_MODE=plugin`, `{NS}=ecc` (the canonical `ecc@ecc` marketplace, 2.0.0+).
+   2. Else if `<claude-home>/plugins/marketplaces/everything-claude-code/` exists → `ECC_MODE=plugin`, `{NS}=everything-claude-code` (pre-2.0.0 marketplace id).
```
[the remark this answers is not in the document]
Comment (gaurav0107) [reply skills/plan-orchestrate/SKILL.md:94]: Holding this as out of scope, with the same reasoning. Claude Code installs marketplace plugins under `<claude-home>/plugins/marketplaces/<marketplace>/` — this repo's own `skills/ecc-recipes/SKILL.md` resolves commands from `$HOME/.claude/plugins/marketplaces/ecc/commands`, and the `install-identifiers` test pins the identifier to `ecc@ecc` → `plugins/marketplaces/ecc/`. A flat `plugins/ecc/` or `plugins/ecc@ecc/` directory is not a layout Claude Code produces, so adding markers for it would document a path that never exists and the legacy fallback you describe cannot occur for a real marketplace install. This change keeps the existing marketplace-path detection contract (which previously only matched the everything-claude-code marketplace) and simply extends it to the renamed `ecc` marketplace per #2316. Glad to revisit in a follow-up if a real flat-install layout is confirmed.

**Comment by gaurav0107** [comment]
Quick note: these CI reds aren't from this PR. `main` itself is failing because its lockfile is out of sync with the eslint 10 bump, so `npm ci` errors before any test runs. #2402 fixes that by resyncing the lockfiles. The bun/pnpm jobs here (including this PR's new tests) are green.

The leftover Windows and Python Tests reds are separate, already-existing issues (a bash-detection test and a missing pyyaml), not related to this change.

**Comment by affaan-m** [comment]
Closing as duplicate: this same fix landed via #2409 (squash `6fac227f` — detect ecc@ecc marketplace + emit ecc: agent prefix), which rewrote the same hunks in skills/plan-orchestrate/SKILL.md and is why this now conflicts. Thanks for the contribution!
